### ISSUE TITLE
Direct Export to Slicers

### ISSUE THREAD

**Issue by Sorcier** [manyfold3d/manyfold#2617]
Currently, there's no direct way to export models from the library to popular slicers like PrusaSlicer or BambuLab Slicer. This requires downloading the model and then manually importing it into the slicer software, which can be a cumbersome extra step.

I would like to see the addition of a functionality that allows users to directly export models from the 3D library to compatible slicers like PrusaSlicer and BambuLab Slicer. Ideally, this could be achieved through integration with their respective APIs or by offering dedicated export buttons that launch the desired slicer with the model pre-loaded.

The current alternative is to download the model and then import it manually into the slicer software. This is not a major issue but adds an unnecessary step to the workflow.

This feature would streamline the workflow for users who want to quickly move from browsing models to slicing them for printing. Direct export to popular slicers would improve user experience and make Manyfold a more attractive platform for 3D printing enthusiasts.

**Comment by Floppy** [comment]
Yes, definitely. I'd love to work out how to do this. I suspect in general it's pretty hard unless the slicers register a special URL handler or something - otherwise it's up to the OS what happens to downloaded files by default. I did wonder if it would need to be a plugin for the slicers to be able to pull models from Manyfold, rather than the other way around. But all that aside, yes, I'd love to get that process smoother.

**Comment by TK-3375** [comment]
Not sure for bambu (makerworld.com has links that open straight in bambustudio) but at least prusaslicer asked me to associate the files so that calls could be made to the slicer on my machine. So it's client dependant I reckon. See printables screenshot below :

![image](https://github.com/user-attachments/assets/7f095159-3f73-462e-9098-3552e8ad1cfb)

**Comment by Floppy** [comment]
Interesting, I hadn't seen the "built-in downloader" thing. I'll take a closer look at what PrusaSlicer and Printables does - maybe there's a custom URL handler. But yeah, in general, it's up to the client browser and OS to decide what to do with a download.

**Comment by Floppy** [comment]
Just noticed that orcaslicer has code in it that has pointed me the required URL structure:

`orcaslicer://open?file=` then the encoded URL to download.

`bambustudio`, `prusaslicer`, and `cura` are done the same way: https://github.com/SoftFever/OrcaSlicer/blob/ca35ba41070aa180bb19a0ba9c4bf74639b50845/src/libslic3r/Utils.hpp#L220

I wonder if the same is true for Lychee, Chitubox, etc etc

**Comment by slitzer** [comment]
I love this Idea, i know it would just need a dir to export the files to and for the likes of OctoPrint it has a setting to auto scan a folder

**Comment by Floppy** [comment]
Thanks! That's really useful investigation, much appreciated!

**Comment by Floppy** [comment]
Hilariously, from the code, it looks like if you include "amazonaws.com" as a meaningless anchor or query argument in your URL, Bambu would go ahead and open it. 

Yay, SECURITY!

**Comment by Floppy** [comment]
Just found https://github.com/shrinerb/shrine/pull/708 which looks like it adds the expiring-URL feature we need in order to do this. Excellent!

**Comment by Floppy** [comment]
OK, Orca and Cura links are now available on the model file detail page, in a dropdown by the button. Next I'll reorganise the download buttons on the listing page and add some user preferences for which method to use by default.

Prusa and Bambu, I've got the URL formats in there, but they won't load from unknown sites, so they're disabled for now. Lychee has a URL handler, but I'm not sure of the format.

**Comment by Floppy** [comment]
Released now. New tickets opened for Prusa and Bambu support, so I can close this one.
